src/pages/home/report/ReportActionItem.tsx

```diff
@@ -586,6 +587,8 @@ function ReportActionItem({
             children = <ReportActionItemBasicMessage message={action.message?.[0]?.text ?? ''} />;
         } else if (action.actionName === CONST.REPORT.ACTIONS.TYPE.UNHOLD) {
             children = <ReportActionItemBasicMessage message={translate('iou.unheldExpense')} />;
+        } else if (action.actionName === CONST.REPORT.ACTIONS.TYPE.DISMISSED_VIOLATION) {
+            children = <ReportActionItemBasicMessage message={ReportActionsUtils.getDismissedViolationMessageText(action.originalMessage)} />;
```

Comment:
I think we should just use the `action.message` we already built in Web-E/Auth instead of duplicating it on the frontend here.

Reply:
This feels like a good thing to get engineering team consensus on. 

also cc @deetergp since you will have to follow a similar pattern for the ESMs doc.

IIRC there was some talk about 🔪 the array format of `action.message` and having Auth queue/return report actions with only the `reportActions.message` (a.k.a. original message). Then the front end could use this however it wants (including building + translating any system messages).

That all makes me wonder whether we have outgrown `report.lastMessageText` field. And if we can stop maintaining the message text code in Auth in general. Though, it's possible we will still need to do have that text built there for push notifications?

And if so, _maybe_ we should be storing the translation texts in Auth.

Reply:
Yeah, these are great questions. I asked about this here: https://expensify.slack.com/archives/C03TQ48KC/p1714167214941959